### PR TITLE
Fix the window doesn't automatically resize.

### PR DESCRIPTION
**Pull request by Horbin-Magician**
Try to fix the issue https://github.com/slint-ui/slint/issues/3911.

In 1.3.0, the function "adjust_window_size_to_satisfy_constraints" is deleted.
I guess that, in Winit 0.29.x, the window will automatically resize, but, at least, it doesn't work on Windows.
So, I try to add this function back.

### CODE REVIEW

**Comment by ogoffart** [comment]
Thanks for the PR!

Did you try it on windows? Does it fix your problem?

File: internal/backends/winit/winitwindowadapter.rs [hunk 1]
```diff
@@ -719,3 +725,28 @@ impl Default for WindowProperties {
...
+    }
+
+    if let Some(max_size) = max_size {
+        let max_size = max_size.cast();
+        window_size.width = window_size.width.min(max_size.width);
+        window_size.height = window_size.height.min(max_size.height);
+    }
+
+    if window_size != winit_window.inner_size() {
+        let _ = winit_window.request_inner_size(window_size);
```
Comment (ogoffart) [inline internal/backends/winit/winitwindowadapter.rs:750]: I think we might need to do something with the return value here.
Other calls to this function store the value. Also we might need to dispatch a resize event

**Comment by Horbin-Magician** [comment]
> Thanks for the PR!
> 
> Did you try it on windows? Does it fix your problem?

I don't seem to be very familiar with git, and I seem to have added a lot of redundant commits, which I hope doesn't bother you.

I have tried it on Windows, ant it does fix my problem.

File: internal/backends/winit/winitwindowadapter.rs (hunk 1 above)
Comment (Horbin-Magician) [reply internal/backends/winit/winitwindowadapter.rs]: Now, I change to dispatch a resize event instead of resizing directly.
Is this a good solution?

**Comment by ogoffart** [comment]
Thanks.

> Now, I change to dispatch a resize event instead of resizing directly. Is this a good solution?

No, what I meant is to take the return value of request_inner_size and only if it did the resize, then dispatch the event, otherwise, we will get the event from winit later.

> I don't seem to be very familiar with git, and I seem to have added a lot of redundant commits, which I hope doesn't bother you.

No problem, i can squash the commits

**Comment by Horbin-Magician** [comment]
> Thanks.
> 
> > Now, I change to dispatch a resize event instead of resizing directly. Is this a good solution?
> 
> No, what I meant is to take the return value of request_inner_size and only if it did the resize, then dispatch the event, otherwise, we will get the event from winit later.
> 
> > I don't seem to be very familiar with git, and I seem to have added a lot of redundant commits, which I hope doesn't bother you.
> 
> No problem, i can squash the commits

In the latest code, I called the 'resize_window' method of `WinitWindowAdapter` instead of using `request_inner_size` method of `winit::window`.

I think the 'resize window' method will help me determine if I need to dispatch the event.
I may not be familiar with the architecture of Slint, maybe you can help me improve it?

**Review by ogoffart — APPROVED** [review]
thanks.  

I'll merge this and test on other platforms.

**Pull request merged**